app/routes.py

```diff
@@ -1,2 +1,234 @@
+        return make_response({"details":"Give video id"}, 400)
+
+    a_video = Video.query.get_or_404(a_video_id)
+    a_customer = Customer.query.get_or_404(a_customer_id)
+
+    if a_video.available_inventory < 1:
+        return make_response({"details":"There are no videos like this available"}, 400)
+
+    a_video.available_inventory -= 1
+    a_customer.videos_checked_out_count += 1
+
+    checked_out_video = Rental(customer_id = a_customer_id, video_id = a_video_id)
+
+    db.session.add(checked_out_video)
+    db.session.commit()
+
+    return jsonify(
+        checked_out_video.rental_response()
+    ), 200
+
+@rental_bp.route("/check-in", methods=["POST"], strict_slashes=False)
+def check_in_video():
+    request_body = request.get_json()
+
+    a_video_id = request_body["video_id"]
+    a_customer_id = request_body["customer_id"]
+    
+    if a_video_id == None or not isinstance(a_video_id, int):
+        return make_response({"details":"Give video id"}, 400)
+
+    if a_customer_id == None or not isinstance(a_customer_id, int):
+        return make_response({"details":"Give video id"}, 400)
+
+    a_video = Video.query.get_or_404(a_video_id)
+    a_customer = Customer.query.get_or_404(a_customer_id)
+        
+    rental = Rental.query.filter_by(customer_id = a_customer_id, video_id = a_video_id).one_or_none()
+    if not rental:
+        return make_response({"details": "This rental doesn't exist"}, 400)
+    checked_in_video = rental
+
+    a_video.available_inventory += 1
+    a_customer.videos_checked_out_count -= 1
+
+    db.session.delete(rental)
+    db.session.commit()
+
+    response = {
+        "customer_id": checked_in_video.customer_id,
+        "video_id": checked_in_video.video_id,
+        "videos_checked_out_count": a_customer.videos_checked_out_count,
+        "available_inventory": a_video.available_inventory
+    }
+
+    return jsonify(
+        response
+    ), 200
+
+@customer_bp.route("/<customer_id>/rentals", methods=["GET"], strict_slashes=False)
+def customer_rentals(customer_id):
+    if not Customer.query.get(customer_id):
+        return make_response({"details": "Customer does not exist"}, 404)
+
+    #use the join query for customer. pattern customer, video, customer
+    all_rentals = db.session.query(Rental).join(Customer, Customer.customer_id==Rental.customer_id)\
+        .join(Video, Video.video_id==Rental.video_id).filter(Customer.customer_id == customer_id).all()
+
+    list_of_rentals = []
+    for rental in all_rentals:
+        #create the return statement and to get the values the set up is the same how rental response is set up. but instead of self use all_rentals. because of the join clause
+        list_of_rentals.append({
+            "release_date": Video.query.get(rental.video_id).release_date,
+            "title": Video.query.get(rental.video_id).title,
+            "due_date": rental.due_date
+        })
+    return jsonify(list_of_rentals), 200
```

Comment:
Requests can slow down the performance of an API, especially when you have hundreds of records you're querying. In this case, you're doing 2 `get()` requests to retrieve a specific attribute of the same video. You can simplify and improve the performance of your API by just doing one `get()` request AND still access the attributes you want. Here is a way to refactor: 

```suggestion
    for rental in all_rentals:
        #create the return statement and to get the values the set up is the same how rental response is set up. but instead of self use all_rentals. because of the join clause
        video = Video.query.get(rental.video_id)
        list_of_rentals.append({
            "release_date": video.release_date,
            "title": video.title,
            "due_date": rental.due_date
        })
    return jsonify(list_of_rentals), 200
```